local_pathfinding/node_mock_global_path.py

```diff
@@ -0,0 +1,449 @@
+                    write=write,
+                    file_path=file_path,
+                )
+            # Check if any waypoints are too far apart
+            elif max(path_spacing) > interval_spacing:
+                self.get_logger().info(
+                    f"Some waypoints in the global path exceed the maximum interval spacing of"
+                    f" {interval_spacing} km. Interpolating between waypoints and generating path"
+                )
+
+                write = self.get_parameter("write")._value
+                if write:
+                    self.get_logger().info("Writing generated path to new file")
+
+                msg = MockGlobalPath.interpolate_path(
+                    global_path=global_path,
+                    interval_spacing=interval_spacing,
+                    pos=pos,
+                    path_spacing=path_spacing,
+                    write=write,
+                    file_path=file_path,
+                )
+
+            else:
+                msg = global_path
+
+            # publish global path
+            self.global_path_pub.publish(msg)
+            self.get_logger().info(
+                f"Publishing to {self.global_path_pub.topic}: {MockGlobalPath.path_to_dict(msg)}"
+            )
+
+    @staticmethod
+    def generate_path(
+        dest: Union[HelperLatLon, list[HelperLatLon]],
+        interval_spacing: float,
+        pos: HelperLatLon,
+        write: bool = False,
+        file_path: str = "",
+    ) -> Path:
+        """Returns a path from the current GPS location to the destination point.
+        Waypoints are evenly spaced along the path according to the interval_spacing parameter.
+        Path does not include pos, but does include dest as the final element.
+
+        If write is True, the path is written to a new csv file in the same directory as file_path,
+        with the name of the original file, appended with a timestamp.
+
+        Args:
+            dest (Union[HelperLatLon, list[HelperLatLon]]):
+                The destination point or partial path
+            interval_spacing (float):
+                The desired distance between waypoints on the path
+            pos (HelperLatLon):
+                The current GPS location
+            write (bool, optional):
+                Whether to write the path to a new csv file, default False
+            file_path (str, optional):
+                The filepath to the csv file containing the global path,
+
+        Returns:
+            Path: The generated path
+        """
+        global_path = Path()
+
+        lat1 = pos.latitude
+        lon1 = pos.longitude
+
+        if isinstance(dest, list):
+            lat2 = dest[0].latitude
+            lon2 = dest[0].longitude
+        else:
+            lat2 = dest.latitude
+            lon2 = dest.longitude
+
+        distance = meters_to_km(GEODESIC.inv(lats1=lat1, lons1=lon1, lats2=lat2, lons2=lon2)[2])
+        n = np.ceil(distance / interval_spacing)
```

Comment:
So if distance is 25 and interval_spacing is 10, we would get n=3 intermediary points. Then global path would be `pos, i1, i2, i3, dest`, and the distance between waypoints would be 6.25.

Am I understanding this correctly? If so, should n be floor not ceil (not accounting for floating point arithmetic)?

Reply:
Yes, I believe you're right, my instinct was to use ceil, but we actually need n intervals so n-1 points is correct.